4 - AnimalShelter/Code/Shelter.cs

```diff
@@ -0,0 +1,106 @@
+﻿using AnimalShelter.Code.Enums;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace AnimalShelter.Code
+{
+    public class Shelter
+    {
+        public Dictionary<Guid?, Animal> StoredAnimals = new Dictionary<Guid?, Animal>();
+        private Guid? id;
+
+        public GetResult AddAnimal(Animal animal)
+        {
+            var result = new GetResult();
+            if (result.Checksupport(animal))
+            {
+                id = Guid.NewGuid();
+                StoredAnimals.Add (id, animal);
+                result.Positiveresult(animal);
+            }
+            return result;     
+        }
+        public GetResult RemoveAnimal(Animal animal)
+        {
+            var result = new GetResult();
+            if (result.Checksupport(animal))
+            {
+                if (ContainsAnimal(animal))
+                {
+                    var key = (from v in StoredAnimals
+                               where v.Value.Equals(animal)
+                               select v.Key).First();
```

Comment:
I would use better variables than like "v". Generally you want to use something that is easy to understand like "animal" or "a" or "sa" in your case.